### PR TITLE
feat: intent extend for staking use cases

### PR DESCRIPTION
**Pull request by hpsing**
* Liquid staking/restaking usecases.

### CODE REVIEW

File: intents.go [hunk 1]
```diff
@@ -11,14 +12,30 @@ import (
 	"github.com/go-playground/validator/v10"
 )
 
+// Type defines the type of asset involved in the transaction.
+type Type string
+
+// Enumeration of possible asset types.
```
Comment (EliasiOfir) [inline intents.go]: IMO it's a state type and not asset type

File: intents.go [hunk 2]
```diff
@@ -11,14 +12,30 @@ import (
...
+	LiquidStake   Type = "STAKE"
+	LiquidReStake Type = "RESTAKE"
 )
 
+// Kind represents the kind of transaction being performed.
+type Kind string // no validate yet, need to have more options later
+
+// Enumeration of possible transaction kinds.
+const (
+	Swap Kind = "swap"
```
Comment (EliasiOfir) [inline intents.go]: what we need "swap" for?

File: intents.go [hunk 3]
```diff
@@ -27,111 +44,122 @@ const (
 	Expired      ProcessingStatus = "Expired"
 	OnChain      ProcessingStatus = "OnChain"
 	Invalid      ProcessingStatus = "Invalid"
-	// evaluate adding specific on chain statuses
 )
 
+// Asset structure for TokenFrom, CurrencyTo, and ReStakeTo
+type Asset struct {
```
Comment (EliasiOfir) [inline intents.go:41]: how do you differ "From" types from "To" types
some can be in both but I guess most will be available only in one side

File: intents.go [hunk 4]
```diff
@@ -27,111 +44,122 @@ const (
...
-	ChainID           *big.Int         `json:"chainId" binding:"required,chain_id"`
+	Sender            string           `json:"sender" binding:"required,eth_addr"`
+	From              Asset            `json:"From"`
+	To                Asset            `json:"To"`
+	PartiallyFillable bool             `json:"PartiallyFillable"`
+	Hash              string           `json:"hash"`
+	CallData          string           `json:"callData"`
+	Status            ProcessingStatus `json:"status" binding:"status"`
+	CreatedAt         int64            `json:"createdAt" binding:"opt_int"`
+	ExpirationAt      int64            `json:"expirationAt" binding:"opt_int"`
```
Comment (EliasiOfir) [inline intents.go]: Fields like PartiallyFillable & ExpirationAt  should be inside "extraData"

File: intents.go (hunk 2 above)
Comment (hpsing) [reply intents.go]: fixed as per new discussion

File: intents.go (hunk 4 above)
Comment (hpsing) [reply intents.go]: we don;t need to maintain an new key:value

**Pull request merged**